packages/features/ee/video/ViewRecordingsDialog.tsx

```diff
@@ -40,116 +41,155 @@ interface GetTimeSpanProps {
   startTime: string | undefined;
   endTime: string | undefined;
   locale: string;
-  isTimeFormatAMPM: boolean;
+  hour12: boolean;
 }
 
-const getTimeSpan = ({ startTime, endTime, locale, isTimeFormatAMPM }: GetTimeSpanProps) => {
+const getTimeSpan = ({ startTime, endTime, locale, hour12 }: GetTimeSpanProps) => {
   if (!startTime || !endTime) return "";
 
   const formattedStartTime = new Intl.DateTimeFormat(locale, {
     hour: "numeric",
     minute: "numeric",
-    hour12: isTimeFormatAMPM,
+    hour12,
   }).format(new Date(startTime));
 
   const formattedEndTime = new Intl.DateTimeFormat(locale, {
     hour: "numeric",
     minute: "numeric",
-    hour12: isTimeFormatAMPM,
+    hour12,
   }).format(new Date(endTime));
 
   return `${formattedStartTime} - ${formattedEndTime}`;
 };
 
+const useRecordingDownload = () => {
+  const [recordingId, setRecordingId] = useState("");
+  const { isFetching, data } = trpc.viewer.getDownloadLinkOfCalVideoRecordings.useQuery(
+    {
+      recordingId,
+    },
+    {
+      enabled: !!recordingId,
+      cacheTime: 0,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+      retry: false,
+      onSuccess: (data) => {
+        if (data && data.download_link) {
+          window.location.href = data.download_link;
+        }
+      },
+    }
+  );
+
+  return {
+    setRecordingId: (newRecordingId: string) => {
+      // may be a way to do this by default, but this is easy enough.
+      if (recordingId === newRecordingId && data) {
```

Comment:
Possibly can be improved with react query caching and refetch, but couldn't figure it out.

Reply:
@emrysal this looks a bit confusing.  
I have set the cacheTime to 0 so even if you click same download recording button onSuccess would get trigger and  set window.location.href = data.download_link

Reply:
we just have to set recording id and that should work